tests/signal/tst_signal.cpp

```diff
@@ -97,170 +97,6 @@ TEST_CASE("Signal connections")
         REQUIRE(lambdaCalled == true);
     }
 
-    SUBCASE("Disconnect Deferred Connection")
```

Comment:
Could you split adding the test and moving the code around the file into separate commits? This way the actual code changes will be comparable in the diff.